### PR TITLE
[Docs] GCP reservation

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/reference/config.rst [hunk 1]
```diff
@@ -99,3 +99,17 @@ Available fields and semantics:
...
       us-east-1: ssh -W %h:%p -p 1234 -o StrictHostKeyChecking=no myself@my.us-east-1.proxy
       us-east-2: ssh -W %h:%p -i ~/.ssh/sky-key -o StrictHostKeyChecking=no ec2-user@<jump server public ip>
+
+  # Advanced GCP configurations (optional).
+  # Apply to all new instances but not existing ones.
+  gcp:
+    # Reserved capacity (optional).
+    #
+    # The specific reservation to be considered when provisioning clusters.
+    # SkyPilot will automatically priortize this reserved capacity (considered as
```
Comment (romilbhardwaj) [inline docs/source/reference/config.rst:109]: nit typo:
```suggestion
    # SkyPilot will automatically prioritize this reserved capacity (considered as
```

File: docs/source/reference/config.rst [hunk 2]
```diff
@@ -99,3 +99,17 @@ Available fields and semantics:
...
+    # zero cost) if the requested resources matches the reservation.
+    # Ref: https://cloud.google.com/compute/docs/instances/reservations-overview#consumption-type
+    specific_reservations:
+      # Only one element is allowed in this list, as GCP disallows multiple
+      # specific_reservations in a single request.
```
Comment (romilbhardwaj) [inline docs/source/reference/config.rst:114]: Curious (and probably more related to the original GCP reservation PR) - is there a particular reason why this is a list? Can it be a str?

File: docs/source/reference/config.rst [hunk 3]
```diff
@@ -99,3 +99,17 @@ Available fields and semantics:
     ssh_proxy_command:
       us-east-1: ssh -W %h:%p -p 1234 -o StrictHostKeyChecking=no myself@my.us-east-1.proxy
       us-east-2: ssh -W %h:%p -i ~/.ssh/sky-key -o StrictHostKeyChecking=no ec2-user@<jump server public ip>
+
+  # Advanced GCP configurations (optional).
+  # Apply to all new instances but not existing ones.
+  gcp:
+    # Reserved capacity (optional).
+    #
+    # The specific reservation to be considered when provisioning clusters.
```
Comment (romilbhardwaj) [inline docs/source/reference/config.rst:108]: Nit
```suggestion
    # The specific reservation to be considered when provisioning clusters on GCP.
```

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll! As someone who hasn't tried this feature, this lgtm. However, someone else who is more familiar should also take a look.

File: docs/source/reference/config.rst (hunk 2 above)
Comment (Michaelvll) [reply docs/source/reference/config.rst:114]: In the future, we expect that we can support multiple `specific_reservations` where we can automatically use all of them by check the number of available reservations in all of them and make use of them. : )

**Pull request merged**